### DISCUSSION TITLE
Integration Tests and CI

### DISCUSSION THREAD

**Discussion by jpriollaud** [ElMassimo/vite_ruby#328] [Q&A] [answered]
My rspec Integration Tests work great locally and the Building with Vite ⚡️ works fine.

But when they run in a Bitbucket Pipelines step (limited to 1 Image, which is Ruby) they fail with:
 Failure/Error: <%= vite_javascript_tag '~/project' %>
     ActionView::Template::Error:
       ❌ The vite binary is not available. Have you installed the npm packages?

Rather than trying to mess with caches and artifacts between a node step and ruby step, can I just configure Vite to not throw errors when its tags do not successfully render?

I feel like I missing something obvious and trying to do it the hard way.

I have read these:
https://github.com/ElMassimo/vite_ruby/discussions/136
https://vite-ruby.netlify.app/guide/development.html#hmr-for-integration-tests-%E2%9C%85

**Comment by ElMassimo** [comment]
Hi there!

> trying to mess with caches and artifacts between a node step and ruby step

I'd recommend you run `assets:precompile` before running integration tests. For that, you'll need to [make sure node is available, and install dependencies](https://github.com/actions/setup-node#caching-global-packages-data).

Not familiar with Bitbucket pipelines, the one image limit might be what's making this tricky.

---

Disabling the JS could cause your app's behavior not to reflect the real behavior in production, I wouldn't recommend doing that for integration tests.

In the future Vite Ruby might provide a test helpers module to ignore assets in controller tests, but again, wouldn't recommend that for integration tests.

**Comment by jpriollaud** [comment]
With no success I tried disabling vite_ruby in the test environment.

I tried modifying vite.json (there's no node!), conditionally disabling bin/vite.

Apparently because of https://github.com/ElMassimo/vite_ruby/blob/41441af83f5cf3b3d02dcdd01fee4aaa1b564a07/vite_ruby/lib/vite_ruby/manifest.rb#L205

`ActionView::Template::Error` `ViteRuby::MissingEntrypointError`

I do not need JS for simple request tests.  So I guess my only remaining options are:
(1) Try to commit the vite-test assets (bad)
(2) Search for a custom Docker container
(3) xit my tests

**Comment by jpriollaud** [comment, marked as answer]
I had to conditionally disable any tags that referenced vite_ruby.

`<%= vite_javascript_tag 'component.jsx' unless Rails.env.test? %>`

It's a shame, because including the vite build in the rspec process (as works locally) was a nice to have.  

Maybe in the future there will better container provisioning in CI, simpler artifact processing, or, in the context of this library, some way of mocking or containerizing the build without node.

@ElMassimo Really appreciate your help.
